src/Symfony/Component/Lock/Store/AbstractStore.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\Lock;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+abstract class AbstractStore implements StoreInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function createLock($resource, $ttl = 300.0)
+    {
+        return new Lock(new Key($resource), $this, $ttl);
```

Comment:
making stores create the lock by passing ``$this`` to the Lock instance makes it very hard to use composition because the Lock instance must always be created by the *outside* store in this case instead of delegating the call to ``createLock`` (otherwise, the lock has the wrong store, as it ends up calling directly the inner store).

IMO, the Store should not be the lock factory at the same time. Mixing these responsibilities is what makes things hard.

We should either have a separate factory (in which the store is injected, and so decoration works fine as the factory would get the outside store) or let the instantiation be done in userland when using the library.